R/plumber-step.R

```diff
@@ -56,21 +56,24 @@ PlumberStep <- R6Class(
       }
     },
     #' @description step execution function
-    #' @param ... additional arguments for step execution
-    exec = function(...) {
-      allArgs <- list(...)
-      args <- getRelevantArgs(allArgs, plumberExpression=private$func)
-
-      hookEnv <- new.env()
+    #' @param req,res Request and response objects created by a Plumber request
+    exec = function(req, res) {
+      hookEnv <- new.env(parent = emptyenv())
 
+      args <- c(
+        # add in `req`, `res` as they have been removed from `req$args`
+        list(req = req, res = res),
+        req$args
+      )
```

Comment:
This should be called right before execution in each of preexecStep, execStep, postexecStep